keymanagementscenarios.md

```diff
@@ -0,0 +1,139 @@
+    - Publisher Admin provides access to Publisher Signer to create or retrieve delegate keys (admins can focus on delegating access, signers can use a portal or APIs to register keys and sign without having to repeatedly engage the admin).
+    - Publisher Signer creates new delegate keys on local machine that chain to root key (the root key is not exposed to the signer).
+    - Publisher Signer runs docker build and notary v2 sign on local machine. Notary v2 client generates signatures with keys on local machine.
+- Signing by Build Machines
+    - Publisher Admin creates new root, uploads root public key to registry, and shares root public key (similar justification to earlier scenarios).
+    - Publisher Admin delegates access for build machine (Publisher Builder and Signer in this context) to create new delegate keys to be used on build machine. The delegate key can be stored on the build machine, USB Token, Network HSM, or Cloud Key Management Service (similar justification to earlier scenarios).
+    - Publisher Admin configures build scripts with credentials and if needed location of delegate keys.
+    - Build machine runs docker build and then notary v2 sign with keys from configured location.
+- Signing by Dedicated Machines
+    - Publisher Admin creates new root, uploads root public key to registry, and shares root public key (similar justification to earlier scenarios).
+    - Publisher Admin delegates access for signing machine (Publisher Signer in this context) to create new delegate keys to be used on build machine. The delegate key can be stored on the build machine, USB Token, Network HSM, or Cloud Key Management Service (similar justification to earlier scenarios).
+    - Publisher Admin configures signing machine with credentials and if needed location of delegate keys.
+    - Build machine (Publisher Builder only in this context) runs docker build and passes objects to be signed to signing host (the build machine does not have access to signing keys. This prevents signing keys from being exposed through potential vulnerabilities in build logic).
+    - Signing host generates signatures with keys in configured location.
+- Signing in Air Gapped Environments with Shared Root 
+    - Publisher Admin creates new root, uploads root public key to registry, and shares root public key (similar justification to earlier scenarios).
+    - Publisher Admin creates new subordinate key in air gapped environment on a host, USB Token, Network HSM, or Cloud Key Management Service chaining to root (one time operation enables the root to be used by runtime environments outside of the air gapped environment to validate signatures generated inside the environment).
+    - Publisher Admin configures credentials for Publisher Signers in air gapped environment to use sub-ordinate key for delegate keys instead of root key (users in air gapped environment do not need to get keys from outside of the environment). 
+
```

Comment:
In general I miss here a scenario that describes managing these keys on a larger scale.

E.g.

The ability for a KeyAdmin to create the repository keys for a team and have the signer keys being selfmanaged by team-members authorized for repositories. Although I might be missing some of the earlier discussions as I also don't see any repository keys discussed anymore (only root and signer). Also not sure if this should be part of the core key management functionality as this might also be something implemented in an overlay integrating with an enterprise SSO to have this level of IAM and RBAC to define who is able to manage what.